src/ducks/auth.ts

```diff
@@ -0,0 +1,355 @@
+import { encode as base64Encode } from "@stablelib/base64";
+import { Networks } from "@stellar/stellar-sdk";
+import { KeyType, ScryptEncrypter } from "@stellar/typescript-wallet-sdk-km";
+import {
+  NETWORKS,
+  SENSITIVE_STORAGE_KEYS,
+  STORAGE_KEYS,
+} from "config/constants";
+import { logger } from "config/logger";
+import { Account, HashKey, KeyPair, TemporaryStore } from "config/types";
+import {
+  deriveKeyFromPassword,
+  encryptDataWithPassword,
+  generateSalt,
+} from "helpers/encryptPassword";
+import { createKeyManager } from "helpers/keyManager/keyManager";
+import { isHashKeyValid } from "hooks/useGetActiveAccount";
+import { t } from "i18next";
+import {
+  dataStorage,
+  secureDataStorage,
+} from "services/storage/storageFactory";
+import { fromMnemonic } from "stellar-hd-wallet";
+import { create } from "zustand";
+
+// Parameters for signUp function
+interface SignUpParams {
+  mnemonicPhrase: string;
+  password: string;
+  imported?: boolean;
+}
+
+// Parameters for storeAccount function
+interface StoreAccountParams {
+  mnemonicPhrase: string;
+  password: string;
+  keyPair: KeyPair;
+  imported?: boolean;
+}
+
+// State slice types
+interface AuthState {
+  network: NETWORKS | null;
+  isAuthenticated: boolean;
+  isLoading: boolean;
+  error: string | null;
+  activeKeyPair: (KeyPair & { accountName: string; id: string }) | null;
+}
+
+interface AuthActions {
+  logout: () => void;
+  signUp: (params: SignUpParams) => void;
+  getIsAuthenticated: () => void;
+  clearError: () => void;
+}
+
+type AuthStore = AuthState & AuthActions;
+
+// Initial state
+const initialState: AuthState = {
+  network: null,
+  isAuthenticated: false,
+  isLoading: false,
+  error: null,
+  activeKeyPair: null,
+};
+
+// Constants
+const HASH_KEY_EXPIRATION_MS = 24 * 60 * 60 * 1000; // 24 hours
+
+/**
+ * Key manager instance for handling cryptographic operations
+ */
+const keyManager = createKeyManager(Networks.TESTNET);
```

Comment:
Could we add a comment here explaining that the Testnet network will be used only as the default network passphrase but that we can actually use this same keyManager instance for any other network?